pkg/util/kms/kms_azure.go

```diff
@@ -0,0 +1,117 @@
+package kms
+
+import (
+	"fmt"
+
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+
+	"github.com/libopenstorage/secrets"
+	"github.com/libopenstorage/secrets/azure"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Azure authentication config options
+const (
+	AzureVaultClientCert = "AZURE_CERT_SECRET_NAME"
+)
+
+// AzureVault is a azure kms driver
+type AzureVault struct {
+	UID string // NooBaa system UID
+}
+
+// NewAzureVault is azure driver constructor
+func NewAzureVault(
+	name string,
+	namespace string,
+	uid string,
+) Driver {
+	return &AzureVault{uid}
+}
+
+//
+// Azure vault specific driver for root master key
+//
+
+// Config returns this driver secret config
+func (*AzureVault) Config(config map[string]string, tokenSecretName, namespace string) (map[string]interface{}, error) {
+	// create a type correct copy of the configuration
+	c := make(map[string]interface{})
+	for k, v := range config {
+		c[k] = v
+	}
+
+	// create TLS files out of secrets
```

Comment:
Give some info about what is needed by the secrets module. What is this cert file and why did we create it also about the other config variables

Reply:
but let's remove all the unneeded as well, like in Get/Set/DeleteContext

Reply:
We get below data as part of `config map[string]string` parameter. This details comes from connectiondetails section of noobaa CR (https://github.com/noobaa/noobaa-operator/blob/906a82032bfc8927145186984034c41577467c9d/pkg/util/kms/kms.go#L99)

ex: configmap data:
```
data:
  AZURE_CERT_SECRET_NAME: azure-ocs-ffwc9o1j
  AZURE_CLIENT_ID: az-client-id1
  AZURE_TENANT_ID: az-tenant-id1
  AZURE_VAULT_URL: az-valut-url1
  KMS_PROVIDER: azure-kv
  KMS_SERVICE_NAME: kms-conn-azure1
```